verticapy/toolbox.py

```diff
@@ -234,6 +247,40 @@ def executeSQL(
     return cursor
 
 
+# ---#
+def extract_col_dt_from_query(query: str, field: str):
+    n, m = len(query), len(field) + 2
+    for i in range(n - m):
+        current_word = query[i : i + m]
+        if current_word.lower() == '"' + field.lower() + '"':
+            i = i + m
+            total_parenthesis = 0
+            k = i + 1
+            while ((query[i] != ",") or (total_parenthesis > 0)) and i < n - m:
+                i += 1
+                if query[i] in ("(", "[", "{"):
+                    total_parenthesis += 1
+                elif query[i] in (")", "]", "}"):
+                    total_parenthesis -= 1
+            return (current_word, query[k:i])
+    return None
+
+
+# ---#
+def extract_compression(path: str):
+    compression = "UNCOMPRESSED"
+    file_extension = path.split(".")[-1].lower()
+    if file_extension[0:2] == "gz":
```

Comment:
Nit: In these use cases, using a lookup table is more efficient and elegant especially when the if-else ladder becomes long.

`lookup_table = {"gz" : "GZIP", "bz" : "BZIP", "lz" : "LZO", "zs" : "ZSTD"}`
`compression = lookup_table[file_extension[0:2]]`

Reply:
Done;